### PR TITLE
Share samples > select project

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

Added a select input to allow the user to select which project they want to copy the sample to.  Once selected all the sample ids within that project are requested from the server and checked against samples that have been selected to share.  If there are existed samples, they get flagged.

## Related issue

Link to the GitHub issue this pull request addresses using the `#issuenum` format. If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist

Things for the developer to confirm they've done before the PR should be accepted:

*   [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
*   [ ] Tests added (or description of how to test) for any new features.
*   [ ] User documentation updated for UI or technical changes.

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectsService.java [hunk 1]
```diff
@@ -240,28 +240,15 @@ public void deleteProject(Long projectId) throws EntityNotFoundException {
...
-
-		Page<Project> paged;
-		if (isAdmin) {
-			paged = projectService.findAllProjects(query, 0, 10, sort);
-		} else {
-			paged = projectService.findProjectsForUser(query, 0, 10, sort);
-		}
-
-		return paged.getContent();
+	public List<Project> getProjectsToShareTo(long currentId) {
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectsService.java:248]: Same as above about renaming the method

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/ProjectsAjaxController.java [hunk 2]
```diff
@@ -74,13 +74,13 @@ public AjaxResponse createNewProject(@RequestBody CreateProjectRequest request,
...
-	 * @param query To search the project name by
-	 * @return List of project the user has rights to that match the query
+	 * @param currentId Current project identifier
+	 * @return {@link List} of {@link Project}s
 	 */
-	@RequestMapping("/user")
-	public ResponseEntity<List<Project>> getUsersProjects(String query) {
-		return ResponseEntity.ok(projectsService.getProjectsForUser(query));
+	@RequestMapping("/samples-share/projects")
+	public List<Project> getProjectsToShareTo(@RequestParam long currentId) {
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/ProjectsAjaxController.java:83]: Can we rename this method? To me the name implies that its going to return projects that a sample **is** going to be shared with not projects that a sample **can** be shared with

File: src/main/webapp/resources/js/apis/projects/projects.js [hunk 3]
```diff
@@ -12,17 +12,15 @@ export const projectsApi = createApi({
   baseQuery: fetchBaseQuery({ baseUrl: URL }),
   tagTypes: ["Projects"],
   endpoints: (build) => ({
-    getProjectsForUser: build.query({
-      query: (query) => ({
-        url: "/user",
-        params: { query },
+    getProjectsToShareTo: build.query({
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/apis/projects/projects.js:15]: This should also be renamed in relation to the above comments

File: src/main/webapp/resources/js/pages/projects/share/index.js [hunk 4]
```diff
@@ -9,28 +14,83 @@ import { render } from "react-dom";
...
-      rowKey={(sample) => `sample-${sample.id}`}
-      columns={[{ title: "Name", dataIndex: "name" }]}
-    />
+    <Form layout="vertical">
+      <Space direction="vertical" style={{ display: "block" }} size="large">
+        <Form.Item label={i18n("ShareSamples.projects")}>
+          <Select
+            style={{ width: `100%` }}
+            loading={projectLoading}
+            options={projects?.map((project) => ({
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/share/index.js:58]: Move this mapping into a function/const

File: src/main/webapp/resources/js/pages/projects/share/index.js [hunk 5]
```diff
@@ -9,28 +14,83 @@ import { render } from "react-dom";
...
+            onChange={updateCurrentSampleIds}
+          />
+        </Form.Item>
+        <Table
+          loading={!originalSamples}
+          dataSource={samples}
+          rowKey={(sample) => `sample-${sample.id}`}
+          columns={[
+            {
+              title: "Name",
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/projects/share/index.js:71]: Internationalize

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Looking good just a few things below

File: src/main/webapp/resources/js/pages/projects/share/index.js (hunk 5 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/share/index.js:71]: This is already changed to a list in it's own component in the next PR

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectsService.java (hunk 1 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectsService.java]: Updated in 5630b02

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/ProjectsAjaxController.java (hunk 2 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/ProjectsAjaxController.java]: Updated in 5630b02

File: src/main/webapp/resources/js/apis/projects/projects.js (hunk 3 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/apis/projects/projects.js]: Updated in 5630b02

File: src/main/webapp/resources/js/pages/projects/share/index.js (hunk 5 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/projects/share/index.js:71]: Sounds good

File: src/main/webapp/resources/js/pages/projects/share/index.js (hunk 4 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/projects/share/index.js]: Updated in b62c182

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**